src/Box.java

```diff
@@ -0,0 +1,54 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class Box<T extends Fruit> {
+    private final List<Fruit> fruits;
+
+    public Box() {
+        fruits = new ArrayList<>();
+    }
+
+    public List<Fruit> getFruits() {
+        return fruits;
+    }
+
+    public void put(T fruit) {
+        if (fruits.isEmpty() || fruit.getClass().equals(fruits.get(0).getClass())) {
+            fruits.add(fruit);
+        } else {
+            throw new IllegalArgumentException("It is prohibited to add different types of fruits");
+        }
+    }
+
+    public void putAll(List<? extends T> fruits) {
+        for (T fruit : fruits) {
+            put(fruit);
+        }
+    }
+
+    public double getWeight() {
+        double weight = 0;
+
+        for (Object fruit : fruits) {
+            if (fruit.getClass().equals(Apple.class)) {
+                weight += 1;
+            } else {
+                weight += 1.5;
+            }
```

Comment:
Class `Box` should know nothing about the type of fruit it is working with. Add a field `weight` to the class `Fruit` and make orange and apple inherit that field through a constructor, and pass the appropriate weight to the constructor of the subclass.